experimental/javareach/cmd/reachable/main.go

```diff
@@ -99,6 +103,14 @@ func enumerateReachabilityForJar(jarPath string) error {
 		return err
 	}
 
+	// Reachability analysis is limited to Maven-built JARs for now.
+	// Check for the existence of the Maven metadata directory.
+	_, err = os.Stat(filepath.Join(tmpDir, MavenDepDirPath))
+	if err != nil {
+		slog.Error("reachability analysis is only supported for JARs built with Maven.")
+		return err
```

Comment:
nit: we should return a more specific error (e.g. a new error type we define) rather than the stat error.